x-pack/plugins/cases/server/client/cases/validators.test.ts

```diff
@@ -0,0 +1,436 @@
+          value: null,
+        },
+      ];
+      expect(() =>
+        validateRequiredCustomFields({
+          requestCustomFields,
+          customFieldsConfiguration,
+        })
+      ).not.toThrow();
+    });
+
+    it('does not throw if there are only optional custom fields in configuration', () => {
+      const customFieldsConfiguration: CustomFieldsConfiguration = [
+        {
+          key: 'first_key',
+          type: CustomFieldTypes.TEXT,
+          label: 'foo',
+          required: false,
+        },
+        {
+          key: 'second_key',
+          type: CustomFieldTypes.TOGGLE,
+          label: 'foo',
+          required: false,
+        },
+      ];
+
+      expect(() =>
+        validateRequiredCustomFields({
+          customFieldsConfiguration,
+        })
+      ).not.toThrow();
+    });
+
+    it('does not throw if the configuration is undefined but no custom fields are in request', () => {
+      expect(() => validateRequiredCustomFields({})).not.toThrow();
+    });
+
+    it('throws if there are missing required custom fields', () => {
+      const requestCustomFields: CaseCustomFields = [
+        {
+          key: 'second_key',
+          type: CustomFieldTypes.TOGGLE,
+          value: true,
+        },
+      ];
+      const customFieldsConfiguration: CustomFieldsConfiguration = [
+        {
+          key: 'first_key',
+          type: CustomFieldTypes.TEXT,
+          label: 'foo',
+          required: true,
+        },
+        {
+          key: 'second_key',
+          type: CustomFieldTypes.TOGGLE,
+          label: 'foo',
+          required: true,
+        },
+      ];
+      expect(() =>
+        validateRequiredCustomFields({
+          requestCustomFields,
+          customFieldsConfiguration,
+        })
+      ).toThrowErrorMatchingInlineSnapshot(`"Missing required custom fields: first_key"`);
+    });
+
+    it('throws if configuration is missing and request has custom fields', () => {
+      const requestCustomFields: CaseCustomFields = [
+        {
+          key: 'first_key',
+          type: CustomFieldTypes.TOGGLE,
+          value: null,
+        },
+      ];
+      expect(() =>
+        validateRequiredCustomFields({
+          requestCustomFields,
+        })
+      ).toThrowErrorMatchingInlineSnapshot(`"No custom fields configured."`);
+    });
+
+    it('throws if configuration has required fields but request has no custom fields', () => {
+      const customFieldsConfiguration: CustomFieldsConfiguration = [
+        {
+          key: 'first_key',
+          type: CustomFieldTypes.TEXT,
+          label: 'foo',
+          required: true,
+        },
+      ];
+      expect(() =>
+        validateRequiredCustomFields({
+          customFieldsConfiguration,
+        })
+      ).toThrowErrorMatchingInlineSnapshot(`"Missing required custom fields: first_key"`);
+    });
+
+    it('throws if request sends a custom field that is not part of the configuration', () => {
```

Comment:
I am wondering if this check exceeds the responsibility of the validation function which is to validate if required fields are present. The `validateCustomFieldKeysAgainstConfiguration` validates for fields that are not part of the configuration. Though they are not connected in this context and are tested in isolation (test in the create/update routes should ensure that both the validation functions are being called). Wdyt?

Reply:
you are right, I'll remove it from the validateRequiredCustomFields fn